### PR TITLE
Split effectful package

### PR DESCRIPTION
**Pull request by mmhat**
As discussed in #20 .

### CODE REVIEW

**Comment by mmhat** [comment]
Note that the `Concurrency` and the `Unlift` benchmarks are _not_ (yet) the ones included in the original `effectful` package since those included the `Async` effect which is not part of the package anymore.

**Comment by arybczak** [comment]
Thanks! For simplicity I'd like the tests and benchmarks to be in the `effectful` package only.

You've also duplicated `README.md`, that isn't going to work. You can have a look at the https://github.com/well-typed/optics repo, it has a similar structure (`optics-core`, `optics-extra` and `optics` as an umbrella package) for inspiration.

**Comment by mmhat** [comment]
Thank you for your feedback!

> For simplicity I'd like the tests and benchmarks to be in the `effectful` package only.

Ah, ok, that makes things a lot easier :+1: 

> You've also duplicated `README.md`

I did that because we include the README in the source distribution and I didn't want to change that. However, I did plan to rewrite it, e.g for the `effectful-core` package:
> # effectful-core
> Some short introduction here. Mention and link to the effectful package and how it relates to it.
> For further information have a look at \<link to README.md in the root of the repository>

But I could as well remove the READMEs from the packages; What do you think?

**Comment by mmhat** [comment]
I removed the READMEs from the packages and updated the package descriptions instead. I started a "Usage" section in the README at the root of the repository (Note: I left a _TBD_ in there; I hope that's ok for now).

As a next step I plan to move the `Resource` effect to an own library; But I will do that in a separate PR. Hence it is ready to be reviewed :slightly_smiling_face:

File: cabal.haskell-ci [hunk 1]
```diff
@@ -9,3 +9,6 @@ ghcup-jobs:     >=9.2
 raw-project
   package effectful
     flags: +benchmark-foreign-libraries
+
+  package effectful-core
```
Comment (arybczak) [inline cabal.haskell-ci:13]: This is not needed anymore.

File: cabal.project [hunk 2]
```diff
@@ -0,0 +1 @@
+packages: */*.cabal
```
Comment (arybczak) [inline cabal.project:1]: Please list cabal files explicitly here.

**Review by arybczak — APPROVED** [review]
Looks good modulo 2 minor comments, thanks :+1:

**Comment by arybczak** [comment]
BTW, `README.md` needs update to a link to the filesystem example.

**Comment by arybczak** [comment]
Uh, it looks that anything pushed to the repo will generate a bunch of conflicts now. I'll try not to push more so you can apply the suggestions and fix existing conflicts.

**Comment by mmhat** [comment]
I resolved the issues you mentioned above. Additionally, I added the `FileSystem` example to the test suite: We don't run the code but we will note when we introduced a breaking change and forgot to update it accordingly.

This PR is good to go :rocket:

**Pull request merged**